.github/workflows/docker-publish.yml

```diff
@@ -23,34 +23,24 @@ jobs:
 
     runs-on: ubuntu-latest
     permissions:
-      contents: read
       packages: write
-      # This is used to complete the identity challenge
-      # with sigstore/fulcio when running outside of PRs.
+      contents: read
+      attestations: write
       id-token: write
 
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v3
-
-      # Install the cosign tool except on PR
-      # https://github.com/sigstore/cosign-installer
-      - name: Install cosign
-        if: github.event_name != 'pull_request'
-        uses: sigstore/cosign-installer@v3.1.1
-      - name: Check install!
-        if: github.event_name != 'pull_request'
-        run: cosign version
-
-      # Workaround: https://github.com/docker/build-push-action/issues/461
-      - name: Setup Docker buildx
-        uses: docker/setup-buildx-action@79abd3f86f79a9d68a23c75a09a9a85889262adf
-
-      # Login against a Docker registry except on PR
-      # https://github.com/docker/login-action
-      - name: Log into registry ${{ env.REGISTRY }}
-        if: github.event_name != 'pull_request'
-        uses: docker/login-action@28218f9b04b4f3f62068d7b6ce6ca5b26e35336c
+      - name: Check out the repo
+        uses: actions/checkout@v4
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
```

Comment:
Mh, why do we need this new QEMU step now?

Reply:
As https://github.com/docker/build-push-action?tab=readme-ov-file#git-context documented

Reply:
From my reading of that, we only need qemu if we cross-compile docker containers:

> [setup-qemu](https://github.com/docker/setup-qemu-action) action can be useful if you want to add emulation support with QEMU to be able to build against more platforms.

We're not doing that though (e.g. we're building natively on x86 and arm64), so everything should work without. Can we try dropping the qemu step and seeing if everything still works?

Reply:
> From my reading of that, we only need qemu if we cross-compile docker containers:
> 
> > [setup-qemu](https://github.com/docker/setup-qemu-action) action can be useful if you want to add emulation support with QEMU to be able to build against more platforms.
> 
> We're not doing that though (e.g. we're building natively on x86 and arm64), so everything should work without. Can we try dropping the qemu step and seeing if everything still works?

I gather this step is necessary for `build-riscv`, not for `build`

Reply:
Alright, I just cleared all the github action runner caches to force it to re-run everything. Let's see how it goes :)

Reply:
How strange, `build-riscv` still works without `qemu-user`. I remember when I tried to build `Dockerfile.riscv64` locally, it cannot proceed if something like `qemu-user-binfmt` is not installed 🤔